src/wrapper.jl

```diff
@@ -319,24 +319,11 @@ function _LGBM_DatasetSetField(ds::Dataset, field_name::String,
     return nothing
 end
 
-function LGBM_DatasetSetField(ds::Dataset, field_name::String, field_data::Vector{Float32})
```

Comment:
I removed these as all the cases are handled by the conversions done in the remaining function. These didn't cover the case of "init_score" which is expected to be Float64 so they were redirecting between label, weight and group field names and using the last function for the init score, which also handles all other field names anyway.
Given my other comments on limitations of restricting the field_data type through signatures, the remaining function if fine to cover all cases.